scripts/coordinate.js

```diff
@@ -213,125 +213,121 @@ FriendlyChat.prototype.removeChats = function() {
   this.messageList.innerHTML = '';
 };
 
-function setAndGetDoWhopDescriptionSession(userSession) {
-  // console.log('Running setAndGetDoWhopDescriptionSession....');
-  var user = person;
-  var userID = person.uid;
-  var currentDoWhopID = userSession.current_dowhop; // This is available from higher scope.
-  var currentTabID = userSession.current_tab;
-  var messagesRef = firebase.database().ref().child('messages/');
-  var sessionRef = database.ref('/session').child(userID);
-  var doWhopSelector = document.getElementById('dowhop-selector-container');
-  var messageList = document.getElementById('messages');
-  var messageInput = document.getElementById('message');
-  var doWhopSelectorDiv = '';
-
-  // Executing functions that are triggered by clicking on a selector block:
-  firebase.database().ref().child('DoWhopDescriptions/' + userSession.current_dowhop).on('value', function(data) {
-    // Weave together header
-    if (data.val()) {
-      var imageUrl =
-        (data.val().downloadURL && data.val().downloadURL.image1) ||
-        data.val().downloadURL ||
-        defaultDoWhopDescriptionImage;
-
-      var doWhopDescriptionTitle = data.val().titleDescription || 'Your DoWhops Will Appear Here';
-
-      // Adding these logic checks so that when users update their information, new times, dates, etc render in 'View':
-      var renderWhenInformation = data.val().whenDescription;
-      var renderWhereInformation = data.val().whereDescription;
-      var renderWhoDescription = data.val().whoDescription;
-      var renderWhoAmIInformation = data.val().whoAmIDescription || '';
-      var whoInformation = data.val().doerDescription; // In progress:  Update with first names dynamically.
-      var renderCreatorIcon = '';
-      var renderDoerIcons = '';
-      var appUsersRef = database.ref('app_users');
-      var doerEmails = whoInformation.split(', ');
-      var creatorEmail = data.val().creatorDescription;
-
-      appUsersRef.once('value')
-      .then(function(snap) {
-        doerUserObjects.length = 0; // Resetting the global variable.
-        creatorUserObjects.length = 0;
-
-        snap.forEach(function(childSnap) {
-          doerEmails.filter(function(doerEmail) {
-            if (doerEmail === childSnap.val().email) {
-              // console.log('it matches!');
-              // console.log(childSnap.val());
-              // console.log('childsnap.val', childSnap.val());
-              doerUserObjects.push(childSnap.val());
-            }
-          });
-          if (creatorEmail === childSnap.val().email) {
-            creatorUserObjects.push(childSnap.val());
+function renderDoWhopMainHeader(userSessionCurrentDoWhop) {
+  var currentDoWhopID = userSessionCurrentDoWhop; // This is available from higher scope.
+  firebase.database().ref().child('DoWhopDescriptions/' + userSessionCurrentDoWhop).once('value').then(function(data) {
```

Comment:
Could be more readable as: 
database.ref('DoWhopDescriptions).child(currentDoWhopID)
  .once('value')
 .then(function(data) {});

Reply:
Added--thanks! 👍